### DISCUSSION TITLE
WATCHTOWER_NO_RESTART doesn't recreate containers after updating

### DISCUSSION THREAD

**Discussion by milindpatel63** [containrrr/watchtower#1440] [General] [answered]
From the description of the argument here, https://containrrr.dev/watchtower/arguments/#without_restarting_containers

It states that the "no restart" option doesn't restart the containers after updating.
But it also doesn't recreate them.
This should be made clear in the docs.
I use systemd to just "start" and "stop" the containers. But i expected watchtower with "WATCHTOWER_NO_RESTART=true" to just recreate the containers after updating and leave them stopped.
Kindly update the docs

**Comment by piksel** [comment, marked as answer]
Yeah, the way it works now doesn't make any sense. I think this is a feature that has been broken by refactoring some time ago.
Currently the `StartContainer` call is the one that creates the new container, *and then starts it*.

This means that the `no-restart` option currently only removes containers which I cannot image is what anyone wants.
